src/components/BookCard/BookCard.tsx

```diff
@@ -0,0 +1,42 @@
+import { BookData } from '../../models';
+import { getBookImgUrl } from '../../data.utils';
+import { useStateFromQuery } from '../../route.utils';
+import { AppLink } from '../AppLink';
+
+interface BookCardProps {
+  book: BookData;
+  parentUrl?: string;
+}
+
+const BookCard = (props: BookCardProps) => {
+  const queryParams = useStateFromQuery();
+  return (
+    <AppLink
+      queryParams={{ ...queryParams, bookId: props.book.id }}
+      className="flex flex-col h-full cursor-pointer"
+    >
+      <div className="grow overflow-hidden flex items-center justify-center p-2">
+        {!props.book.coverId ? (
```

Comment:
what if the `book` is `null`? What if the bood does not have `coverId`? Add a check please